### PR TITLE
Support imported files in thrift-gen

### PR DESCRIPTION
**Pull request by thanodnl**
This PR adds support for imported thrift files during the code generation phase.

`GoUnusedProtection__` is the same trick as used in [`thrift-gen`](https://github.com/uber/tchannel-go/blob/v1.0.9/thrift/thrift-gen/tchannel-template.go#L22)

### CODE REVIEW

File: test/remoteservice/remoteservice.thrift [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
+
+// the unused file contains types that are not used in service endpoints. These
+// types might actually be used in embedded structs here, but since the
+// generated code could potentially contain the import of an unused package in
+// the RingpopAdapter. To prevent this the generator uses `GoUnusedProtection__`
+include "./unused.thrift"
+
 service RemoteService {
-  void RemoteCall(1: string name)
+  void RemoteCall(1: shared.UUID id)
```
Comment (mennopruijssers) [inline test/remoteservice/remoteservice.thrift]: Why change it from `name` to `id`? To have less changes, you could keep it name and define a `Name` type in shared.thrift (`typedef string Name`)?

**Comment by mennopruijssers** [comment]
I'm fine with changing the name to an uuid although it implies more changes.
Feel free to change it or keep it like this (LGTM)

**Comment by mennopruijssers** [comment]
🎉

**Comment by thanodnl** [comment]
The failure is a flap:

```
--- FAIL: TestProtocolStats (5.09s)
	Error Trace:	stats_test.go:89
	Error:		Should not be zero, but was 0
		
	Error Trace:	stats_test.go:90
	Error:		Should not be zero, but was 0
		
--- FAIL: TestStatsTestSuite (5.21s)
FAIL
FAIL	github.com/uber/ringpop-go/swim	9.984s
```

Nothing changed that would impact a failure in `swim` with this change. Re-running test.

File: test/remoteservice/unused.thrift [hunk 2]
```diff
@@ -0,0 +1 @@
+typedef string UnusedType
```
Comment (CorgiMan) [inline test/remoteservice/unused.thrift]: This isn't used anywhere, what is it for?

**Comment by CorgiMan** [comment]
Looks good

**Pull request merged**